File: tp3/src/Inicializador.java
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.TreeMap;

import com.sun.org.apache.bcel.internal.generic.UnconditionalBranch;


public class Inicializador {

	/**
	 * @param args
	 * @throws ExcepcionProductoInsat
	 * @throws ClienteNuloTelefono
	 * @throws ExcepcionOrigenDesconocido
	 * @throws PedidoNulo
	 * @throws ProductosVacios
	 */
	public static void main(String[] args) throws ProductosVacios, PedidoNulo, ExcepcionOrigenDesconocido, ClienteNuloTelefono, ExcepcionProductoInsat {
		TipoProducto pizza= new TipoProducto("pizza",true,true);
		TipoProducto empanada= new TipoProducto("empanada",true,true);
		TipoProducto coca= new TipoProducto("coca",false,false);
		TipoProducto birra= new TipoProducto("cerveza",false,false);
		Insumo bollo = new Insumo(10,3,"bollo");
		Insumo muzza = new Insumo(13,2,"muzza");
		Insumo tapa = new Insumo(15,3,"tapa");
		Insumo carne = new Insumo(34,5,"carne");
		Insumo quilmes = new Insumo(23,3,"quilmes");
		Insumo pepsi = new Insumo(123,23,"pepsi");
		List<Insumo> insumosMuzza = new LinkedList<Insumo>();
		insumosMuzza.add(bollo);
		insumosMuzza.add(muzza);
		Producto pizzMuzza = new Producto("pizzaMuzza",10,13.3,2.3,pizza,insumosMuzza);
		System.out.println(Producto.allInstances().toString());

		//ahora vamos a armar un fraccionador
		Map<TipoProducto,Integer> t = new TreeMap<TipoProducto,Integer>();
		t.put(pizza, 1);
		t.put(empanada, 4);
		FraccionadorDeHorno frPizzero = new FraccionadorDeHorno(t);
		FraccionadorDeHorno frEmpanadero = new FraccionadorDeHorno(t);
		//armamos 2 hornos
		Horno hpizzero = new Horno(10,frPizzero);
		Horno hempanadero = new Horno(10,frEmpanadero);
		// armamos los avisos
		AvisoEmpanadero avE = new AvisoEmpanadero(null);
		AvisoPizzero avP = new AvisoPizzero(null);
		//ahora los preparadores
		PreparadorEspecializado prepPizzero= new PreparadorEspecializado(avP);
		PreparadorEspecializado prepEmpanadero= new PreparadorEspecializado(avE);
		//coordinadorDeCocina
		DespachadorDePreparacion dep=null;
		CoordinadorDeCocina coordC = new CoordinadorDeCocina(null,null,null);
		//Despachador
		dep = new DespachadorDePreparacionStandard(prepPizzero,prepEmpanadero,coordC,pizza,empanada);
		coordC.setDespachadorDePreparacion(dep);
		//controladorDeIngreso
        ControladorDeIngreso contIng = new ControladorDeIngresoStandard(coordC);
        coordC.setControladorDeIngreso(contIng);
        //controladorDelistos
        ControladorDeListos contList = new ControladorDeListos();
        //CoordinadorDePedidos
        CoordinadorDePedidos coordP = new CoordinadorDePedidos(null,null,contList);
        coordC.setCoordinadorDePedidos(coordP);
        //controladorDePreIngreso
        ControladorDePreIngreso contPreIng = new ControladorDePreIngreso(coordP,contIng);
        coordP.setControladorDePreIngreso(contPreIng);
        //CalculadorDePrecios
        CalculadorDePrecios calcP = new CalculadorDePreciosStandard();
        //AsignadorDeHorno
        AsignadorDeHorno asigH = new AsignadorDeHornoStandard(hpizzero,hempanadero,pizza,empanada);
        //EstimadorDeTiempos
        EstimadorDeTiempos estT = new EstimadorStandard(pizza,empanada);
        //ControladorDeStock
        ControladorDeStock contS = new ControladorDeStockStandard();
        //GenedarodDePedidos
        GeneradorDePedidos genP = new GeneradorDePedidosStandard(calcP,estT,contS,asigH);
        coordP.setGeneradorDePedidos(genP);
        List<Producto> lp= new LinkedList<Producto>();
        lp.add(pizzMuzza);
        lp.add(pizzMuzza);
        coordP.ingresarPedido(null, lp, "contado", "mostrador", 0);
        coordP.ingresarPedido(null, lp, "contado", "mostrador", 0);
        coordP.ingresarPedido(null, lp, "contado", "mostrador", 0);


	}

}

File: tp3/src/inicializador.py

#!/usr/bin/python

from cocina import *
from gestion import *
from creacion import *

import pickle

class Pizzeria:
    def __init__(self, f=None):
        if not f is None:
            dic=self.cargarDiccionario(f)
            self.cargarInsumos(dic)
            self.cargarProductos(dic)
            self.cargarTipoProducto(dic)
            self.cargarClientes(dic)
        
        #ahora vamos a armar un fraccionador
        t = {}
        t[self.pizza]=1
        t[self.empanada]= 4
        frPizzero =  FraccionadorDeHorno(t)
        frEmpanadero =  FraccionadorDeHorno(t)
        #armamos 2 hornos
        self.hpizzero =  Horno(10,frPizzero,"Horno Pizzero")
        self.hempanadero =  Horno(10,frEmpanadero,"Horno Empanadero")
        # armamos los avisos
        avE =  AvisoEmpanadero(None)
        avP =  AvisoPizzero(None)
        #ahora los preparadores
        self.prepPizzero=  PreparadorEspecializado(avP)
        self.prepEmpanadero=  PreparadorEspecializado(avE)
        #coordinadorDeCocina
        dep=None
        self.coordC =  CoordinadorDeCocina(None,None,None)
        #Despachador
        dep =  DespachadorDePreparacionStandard(self.prepPizzero,self.prepEmpanadero,self.coordC,self.pizza,self.empanada)
        self.coordC.setDespachadorDePreparacion(dep)
        avE.setDestinatario(dep)
        avP.setDestinatario(dep)
        #controladorDeIngreso
        self.contIng =  ControladorDeIngresoStandard(self.coordC)
        self.coordC.setControladorDeIngreso(self.contIng)
        #controladorDelistos
        self.contList =  ControladorDeListos()
        #CoordinadorDePedidos
        self.coordP =  CoordinadorDePedidos(None,None,self.contList)
        self.coordC.setCoordinadorDePedidos(self.coordP)
        #controladorDePreIngreso
        contPreIng =  ControladorDePreIngreso(self.coordP,self.contIng)
        self.coordP.setControladorDePreIngreso(contPreIng)
        #CalculadorDePrecios
        calcP =  CalculadorDePreciosStandard()
        #AsignadorDeHorno
        self.asigH =  AsignadorDeHornoStandard(self.hpizzero,self.hempanadero,self.pizza,self.empanada)
        #EstimadorDeTiempos
        estT =  EstimadorStandard(self.pizza,self.empanada)
        #ControladorDeStock
        self.contS =  ControladorDeStockStandard()
        #GenedarodDePedidos
        genP =  GeneradorDePedidosStandard(calcP,estT,self.contS,self.asigH)
        self.coordP.setGeneradorDePedidos(genP)
   
    def cargarInsumos(self,dic):
        self.insumos =dic["Insumos"]
    
    def cargarProductos(self,dic):
        self.productos=dic["Productos"]
    
    def cargarTipoProducto(self,dic):
        tipoproductos = dic['TiposProducto']
        self.pizza=tipoproductos['pizza']
        self.empanada=tipoproductos['empanada']
        self.coca=tipoproductos['coca']
        self.birra=tipoproductos['birra']
                
    def cargarClientes(self,dic):
        self.clientes=dic["Clientes"]
        
    def cargarDiccionario(self,f):
        dic=pickle.load(f)
        return dic
    
    def getContIng(self):
        return self.contIng
    
    def getContListos(self):
        return self.contList
    
    def getCoordC(self):
        return self.coordC
        
    def getCoordP(self):
        return self.coordP
    
    def getPreparadorEmpanadero(self):
        return self.prepEmpanadero
    
    def getPreparadorPizzero(self):
        return self.prepPizzero
    
    def getContStock(self):
        return self.contS
    
    def getAsignador(self):
       return self.asigH 
    
    def getHornoE(self):
	return self.hempanadero
   
    def getHornoP(self):
        return self.hpizzero
        

if __name__ == '__main__':
            
    fi = open("datos.pyp", 'rb')        
    sis = Pizzeria(f=fi)
    print Producto.allInstances()
